### ISSUE TITLE
question on dataset on the fly

### ISSUE THREAD

**Issue by MiqG** [gagneurlab/scooby#2]
Hi! Congrats to all the authors for developing such a cool model.

I have been developing a similar approach for cell-specific splicing prediction and one of the major hurdles has been feeding the data to the model. 

In my case, I have been preprocessing and storing all input data (sequence chunks + omic profiles) in giant .h5 files, which take quite some time to make but are read very fast during training without any appreciable latency between batches. However, every time one makes a mistake or wants to add some new data need to make .h5 files again...

I really like your "onTheFly" classes as a great solution to this problem! I was wondering two things:
- where can I find the package "polya_project.data" to import the FastaInterval and GenomeIntervalDataset classes? (see `scdata.py`: https://github.com/gagneurlab/scooby/blob/058e6f65a54b3f473e287aed5b0bda3b6e711b67/data/scdata.py#L10C1-L10C68) . Googling it I have seen these classes are also found in the [`enformer-pytorch`](https://github.com/lucidrains/enformer-pytorch) repo but I wanted to make sure they are the same before recycling them
- do you experience any latency time between GPU usage and data loading between batches during training?

Thanks in advance! Best,

Miquel

**Comment by johahi** [comment]
Hi Miquel, thanks for your interest! 

* We have just pushed a tiny fix to the enformer-pytorch repository, and that's indeed the file we're referring to here. We'll have a cleaned-up version of the code out very soon - just recycle the classes from there as you propose until then.
* Regarding your second question, it depends a bit on how many cell targets (observations) you load per batch. We tried our best to optimize this and for some hundreds of cells (and their stranded RNA + ATAC tracks), this decently overlaps with GPU usage and gradient sync across GPUs for our setup. Broadly, we can load targets for 1k cells within seconds. 

I have worked with [webdataset](https://github.com/webdataset/webdataset) before, maybe this also suits your purpose if you want to add more cells/assays.

Best, Johannes
